analyses/focal-cn-file-preparation/01-add-ploidy-cnvkit.Rmd

```diff
@@ -54,21 +54,59 @@ add_autosomes_df <- add_ploidy_df %>%
+
+
+  copy.num > (0.5 * tumor_ploidy) & case_when (cohort == 'TARGET' & germline_sex_estimate == NA ~ case_when (samples_superset$germline_sex_estimate.x == NA ~ reported_gender, TRUE ~ samples_superset$germline_sex_estimate.x), TRUE ~ germline_sex_estimate) == "Male" ~ "gain",
+
+  copy.num == (0.5 * tumor_ploidy) & case_when (cohort == 'TARGET' & germline_sex_estimate == NA ~ case_when (samples_superset$germline_sex_estimate.x == NA ~ reported_gender, TRUE ~ samples_superset$germline_sex_estimate.x), TRUE ~ germline_sex_estimate) == "Male" ~ "neutral",
+
+
+  
+
+  # there are some instances where there are chromosome Y segments are in
+  # samples where the germline_sex_estimate is Female
+
+  chrom == "chrY" & copy.num > 0 & case_when ( cohort == "PBTA" & germline_sex_estimate == NA ~ samples_superset$germline_sex_estimate.x , TRUE ~ germline_sex_estimate) == "Female" ~ "unknown",
+
+  chrom == "chrY" & copy.num > 0 & case_when (cohort == 'TARGET' & germline_sex_estimate == NA ~ case_when (samples_superset$germline_sex_estimate.x == NA ~ reported_gender, TRUE ~ samples_superset$germline_sex_estimate.x), TRUE ~ germline_sex_estimate) == "Female" ~ "unknown",
+
+  chrom == "chrY" & copy.num == 0 & case_when ( cohort == "PBTA" & germline_sex_estimate == NA ~ samples_superset$germline_sex_estimate.x , TRUE ~ germline_sex_estimate) == "Female"  ~ "neutral",
+
+  chrom == "chrY" & copy.num == 0 & case_when (cohort == 'TARGET' & germline_sex_estimate == NA ~ case_when (samples_superset$germline_sex_estimate.x == NA ~ reported_gender, TRUE ~ samples_superset$germline_sex_estimate.x), TRUE ~ germline_sex_estimate) == "Female" ~ "neutral",
+
+
+
+
+  # mirroring controlfreec, X treated same as autosomes
+
+  chrom == "chrX" & copy.num < tumor_ploidy & case_when ( cohort == "PBTA" & germline_sex_estimate == NA ~ samples_superset$germline_sex_estimate.x , TRUE ~ germline_sex_estimate) == "Female" ~ "loss",
+  chrom == "chrX" & copy.num < tumor_ploidy & case_when (cohort == 'TARGET' & germline_sex_estimate == NA ~ case_when (samples_superset$germline_sex_estimate.x == NA ~ reported_gender, TRUE ~ samples_superset$germline_sex_estimate.x), TRUE ~ germline_sex_estimate) == "Female" ~ "loss",
+
+  chrom == "chrX" & copy.num > tumor_ploidy & case_when ( cohort == "PBTA" & germline_sex_estimate == NA ~ samples_superset$germline_sex_estimate.x , TRUE ~ germline_sex_estimate) == "Female" ~ "gain",
+  chrom == "chrX" & copy.num > tumor_ploidy & case_when (cohort == 'TARGET' & germline_sex_estimate == NA ~ case_when (samples_superset$germline_sex_estimate.x == NA ~ reported_gender, TRUE ~ samples_superset$germline_sex_estimate.x), TRUE ~ germline_sex_estimate) == "Female" ~ "gain",
+
+  chrom == "chrX" & copy.num == tumor_ploidy & case_when ( cohort == "PBTA" & germline_sex_estimate == NA ~ samples_superset$germline_sex_estimate.x , TRUE ~ germline_sex_estimate) == "Female" ~ "neutral"
```

Comment:
I also think the easier way to handle this would be as @runjin326 suggested to first update the germline_sex_estimate to reported_gender where required ( where WXS germline_sex_estimate is NA) and then follow the original logic to update status. 

Here I see that a comma is also missing so that could be the source of the original error? Could you post the error here?

Reply:
Thank you @kgaonkar6 
The code works now and performs as expected.